portfolio/src/main/webapp/scripts/survey.js

```diff
@@ -0,0 +1,76 @@
+const subgenres = ['Body Horror', 'Slasher', 'Monster', 'Paranormal', 'Psychological Thriller'];
+const optionsContainer = document.getElementById('options-container');
+createSurvey();
+drawChart(); 
+
+/*
+ * Creates and adds individual option elements to the DOM
+ */ 
+function createSurvey() {
+  for (const optionName of subgenres) {
+    let option = document.createElement('input');
+    option.setAttribute('type', 'radio'); 
+    option.setAttribute('name', 'option');
+    option.setAttribute('value', optionName);
+    let optionLabel = document.createElement('label');
+    optionLabel.setAttribute('for', optionName);
+    optionLabel.textContent = optionName; 
+    optionsContainer.appendChild(option); 
+    optionsContainer.appendChild(optionLabel); 
+  }
+}
+
+/*
+ * Get the user's reponse to the survey
+ */  
+function getUserResponse() {
+  let options = document.getElementsByName('option'); 
+  for (const option of options) {
+    if (option.checked) {
+      addVote(option.value); 
+    }
+  }
+}
+
+/*
+ * Add the user's vote to the database 
+ */ 
+function addVote(option) {
+  const params = new URLSearchParams(); 
+  params.append('subgenre', option); 
+  fetch('/survey', {
+    method: 'POST',
+    body: params
+  }).catch(err => {
+    alert('There was an error trying to add your vote. Try checking your internet connection and refreshing the page.'); 
+  });
+  drawChart(); 
+}
+
+
+google.charts.load('current', {'packages':['corechart']});
+/*
+ * Fetches subgenre data and uses it to create a chart. 
+ */
+async function drawChart() {
+  try {
+    const response = await fetch('/survey');
+    const surveyResponse = await response.json(); 
```

Comment:
Don't need to await here and above.  awaiting once fulfills the promise.

Reply:
Removed unnecessary awaits